### PR TITLE
clang-format files for iOS, shared and experimental

### PR DESCRIPTION
**Pull request by licanhua**
This PR only format files for ios, shared and experimental folders.
Pipeline is not enforced to detect the format issue yet.

`uwp` is excluded from the format because of #6794 which may introduce massive merge conflicts if I format them.
 
Format command: 
```
PowerShell.exe -ExecutionPolicy Bypass scripts\FormatSource.ps1 -ModifiedOnly $False
```
or
```
npm run format
```

Verify command
```
PowerShell.exe -ExecutionPolicy Bypass scripts\FormatSource.ps1 -ModifiedOnly $False -Verify $True
```
or
```
npm run verify
```

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/6851)

### CODE REVIEW

File: source/experimental/ConsoleRenderer/ConsoleRenderer/ConsoleRender.cpp [hunk 1]
```diff
@@ -332,23 +334,23 @@ void RenderBorder(std::vector<std::string> &output, size_t columnWidth)
...
-void RenderToConsole(std::shared_ptr<AdaptiveCards::AdaptiveCard> card, size_t columnWidth, std::vector<std::string> &output)
+void RenderToConsole(std::shared_ptr<AdaptiveCards::AdaptiveCard> card, size_t columnWidth, std::vector<std::string>& output)
 {
     std::vector<std::string> outputVector;
     auto body = card->GetBody();
-    RenderElements(body, columnWidth-4, 1, 1, output); // save 4 width for border
+    RenderElements(body, columnWidth - 4, 1, 1, output); // save 4 width for border
     RenderBorder(output, columnWidth);
-    //printf("\n\n\n");
+    // printf("\n\n\n");
```
Comment (almedina-ms) [inline source/experimental/ConsoleRenderer/ConsoleRenderer/ConsoleRender.cpp:347]: >// printf("\n\n\n"); [](http://example.com/codeflow?start=4&length=20)

lots of comments that we could erase, as this is a formatting PR maybe we could do it in a different PR?

Comment (almedina-ms) [reply source/experimental/ConsoleRenderer/ConsoleRenderer/ConsoleRender.cpp:347]: this is on the console renderer so maybe this is not important

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRBaseCardElementRenderer.mm [hunk 2]
```diff
@@ -17,9 +17,9 @@ + (ACRCardElementType)elemType
 /* not implemented ATM */
 - (void)setSeparationConfig:(UIView *)viewGroup
 {
-    //SeparationStyle separationStyle,
-    //SeparationConfig defaultSeparationConfig,
-    //SeparationConfig strongSeparationConfig,
+    // SeparationStyle separationStyle,
+    // SeparationConfig defaultSeparationConfig,
+    // SeparationConfig strongSeparationConfig,
     //(Boolean) horizontalLine)
```
Comment (almedina-ms) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRBaseCardElementRenderer.mm:23]: this looks like this should be filed as a bug

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/include/ACOActionOverflow.h [hunk 3]
```diff
@@ -1 +1 @@
-../ACOActionOverflow.h
\ No newline at end of file
+../ ACOActionOverflow.h
```
Comment (almedina-ms) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/include/ACOActionOverflow.h:1]: is the space between the slash "/" and the file name correct?

File: source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer/ViewController.m [hunk 4]
```diff
@@ -168,10 +168,9 @@ - (IBAction)applyText:(id)sender
 - (IBAction)deleteAllRows:(id)sender
 {
     [(ACRChatWindow *)self.chatWindow.dataSource deleteAllRows:self.chatWindow];
-    
+
     // clean the retrieved inputs
-    if ([self appIsBeingTested])
-    {
+    if ([self appIsBeingTested]) {
```
Comment (almedina-ms) [inline source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer/ViewController.m:173]: >{ [](http://example.com/codeflow?start=33&length=1)

this looks wrong, all other open curly braces are added in a new line, is this only an if/else rule?

File: source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizerTests/Action.ExecuteTest.mm [hunk 5]
```diff
@@ -82,24 +82,23 @@ - (void)testRefreshInterface
     executeAction->SetVerb("doStuff");
     const std::string dataString = "{\"HereIs\" :  \"Some Data\"}";
     executeAction->SetDataJson(dataString);
-    ACORefresh *refresh = [[ACORefresh alloc] init:[ACOBaseActionElement getACOActionElementFromAdaptiveElement:executeAction] userIds:@[@"user1", @"user2", @"user3"]];
+    ACORefresh *refresh = [[ACORefresh alloc] init:[ACOBaseActionElement getACOActionElementFromAdaptiveElement:executeAction] userIds:@[ @"user1", @"user2", @"user3" ]];
```
Comment (almedina-ms) [inline source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizerTests/Action.ExecuteTest.mm:85]: >  [](http://example.com/codeflow?start=166&length=1)

extra space?

File: source/experimental/ConsoleRenderer/ConsoleRenderer/ConsoleRender.cpp (hunk 1 above)
Comment (licanhua) [reply source/experimental/ConsoleRenderer/ConsoleRenderer/ConsoleRender.cpp:347]: Yes. this PR only do formatting files with tools, and will have no manual code change on formatted files.

File: source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer/ViewController.m (hunk 4 above)
Comment (licanhua) [reply source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer/ViewController.m:173]: I believe it's controlled by `BraceWrappingAfterControlStatementStyle AfterControlStatement Wrap control statements (if/for/while/switch/..).`
and applied to other controls statement. If not, please let me know.

It's documented in https://clang.llvm.org/docs/ClangFormatStyleOptions.html.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/include/ACOActionOverflow.h (hunk 3 above)
Comment (licanhua) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/include/ACOActionOverflow.h:1]: I don't know about it. Can you do a testing on your local machine? I didn't see errors in build pipeline, and I guess it's supported.

File: source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizerTests/Action.ExecuteTest.mm (hunk 5 above)
Comment (licanhua) [reply source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizerTests/Action.ExecuteTest.mm:85]: Is it a list? if yes, it's controlled by SpacesInContainerLiterals and was expected.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRBaseCardElementRenderer.mm (hunk 2 above)
Comment (licanhua) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRBaseCardElementRenderer.mm:23]: I don't know why it's not `// (Boolean) horizontalLine)`, and I didn't see a rule to make it remove the space.
I assume it's a bug of clang-format itself, and I would just ignore it.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRIBaseCardElementRenderer.h [hunk 6]
```diff
@@ -21,12 +21,12 @@
...
          hostConfig:(ACOHostConfig *)acoConfig;
 @optional
-/// override this method for custom styling 
+/// override this method for custom styling
 /// not all renderers supports it
 - (void)configure:(UIView *)view
-         rootView:(ACRView *)rootView
-  baseCardElement:(ACOBaseCardElement *)acoElem
-       hostConfig:(ACOHostConfig *)acoConfig;
+           rootView:(ACRView *)rootView
```
Comment (jwoo-msft) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRIBaseCardElementRenderer.h:27]: nit: this one looks weird

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/include/ACOActionOverflow.h (hunk 3 above)
Comment (jwoo-msft) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/include/ACOActionOverflow.h:1]: yes, this is a bug. I tested it. the symbolic link was broken.

**Review by jwoo-msft — CHANGES_REQUESTED** [review]
Thank you for this PR.
There are many files that had slipped through the clang-format
I'm now a firm advocate for having a format enforcement pipeline check.
The changes look great except those symbolic links, all headers inside `\include`.
Maybe we should not go into the `\include` since they are generated files anways.
with the added space, the links were broken.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/include/ACOActionOverflow.h (hunk 3 above)
Comment (licanhua) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/include/ACOActionOverflow.h]: Thanks. I revert the change on include folder. Do you know why the pipeline doesn't catch the problem?

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRIBaseCardElementRenderer.h (hunk 6 above)
Comment (licanhua) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRIBaseCardElementRenderer.h:27]: didn't get it. what's the problem

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/include/ACOActionOverflow.h (hunk 3 above)
Comment (jwoo-msft) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/include/ACOActionOverflow.h]: these files are used by swift package manager. and not directly by ci pipeline. it's a good idea, i just added cocoapod spec check, we can add this spm check in next eim.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRIBaseCardElementRenderer.h (hunk 6 above)
Comment (jwoo-msft) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRIBaseCardElementRenderer.h:27]: i don't know if you can fix it. if you can't, just leave it for now. you're using the clang-format config in the ios folder, right?

the clang-format usually lines up the ':'.
on that line, I couldn't see why it won't line them together.

File: source/shared/cpp/ObjectModel/ActionParserRegistration.h [hunk 7]
```diff
@@ -7,47 +7,50 @@
 
 namespace AdaptiveCards
 {
-    class BaseActionElement;
-    class ParseContext;
+class BaseActionElement;
```
Comment (jwoo-msft) [inline source/shared/cpp/ObjectModel/ActionParserRegistration.h:10]: nit: is this something agreed to change? looks different from what we had before.

File: source/shared/cpp/ObjectModel/BackgroundImage.cpp [hunk 8]
```diff
@@ -117,7 +117,8 @@ std::shared_ptr<BackgroundImage> BackgroundImage::Deserialize(const Json::Value&
 
     image->SetUrl(ParseUtil::GetString(json, AdaptiveCardSchemaKey::Url, true));
 
-    image->SetFillMode(ParseUtil::GetEnumValue<ImageFillMode>(json, AdaptiveCardSchemaKey::FillMode, ImageFillMode::Cover, ImageFillModeFromString));
+    image->SetFillMode(ParseUtil::GetEnumValue<ImageFillMode>(
```
Comment (jwoo-msft) [inline source/shared/cpp/ObjectModel/BackgroundImage.cpp:120]: nit: line break seems to change config to config. :-)

File: source/shared/cpp/ObjectModel/BaseCardElement.cpp [hunk 9]
```diff
@@ -27,11 +27,12 @@ BaseCardElement::BaseCardElement(CardElementType type) :
 
 void BaseCardElement::PopulateKnownPropertiesSet()
 {
-    m_knownProperties.insert({AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::Height),
-                              AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::IsVisible),
-                              AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::MinHeight),
-                              AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::Separator),
-                              AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::Spacing)});
+    m_knownProperties.insert(
```
Comment (jwoo-msft) [inline source/shared/cpp/ObjectModel/BaseCardElement.cpp:30]: nit: max line width has changed from the old config? i have now problems with the changes, but i'm wondering if this's agreed.

File: source/shared/cpp/ObjectModel/ChoiceSetInput.cpp [hunk 10]
```diff
@@ -9,8 +9,7 @@
 using namespace AdaptiveCards;
 
 ChoiceSetInput::ChoiceSetInput() :
-    BaseInputElement(CardElementType::ChoiceSetInput), m_wrap(false), m_isMultiSelect(false),
-    m_choiceSetStyle(ChoiceSetStyle::Compact)
+    BaseInputElement(CardElementType::ChoiceSetInput), m_wrap(false), m_isMultiSelect(false), m_choiceSetStyle(ChoiceSetStyle::Compact)
```
Comment (jwoo-msft) [inline source/shared/cpp/ObjectModel/ChoiceSetInput.cpp:12]: nit: inconsitency; why does this one make a longer line? the previous changes in this PR took the longer ones to multiple lines.

**Review by jwoo-msft — APPROVED** [review]
iOS changes make sense to me since there were many files that were not formatted especially in visualizer and unit tests codes.
C++ changes look great, too.
But i'm wondering why we have to make these changes in C++.
old and new ones look equally formatted properly and well.
for example,
removing spaces after names namespace
removing spaces in set and map intialization. 
changing max-width of the line from a value to b value.
i approve this PR for iOS, but I defer others to comments on C++.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRIBaseCardElementRenderer.h (hunk 6 above)
Comment (licanhua) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRIBaseCardElementRenderer.h:27]: yes. I'm using clang-format config in the ios folder and didn't do any change on it.

File: source/shared/cpp/ObjectModel/ActionParserRegistration.h (hunk 7 above)
Comment (licanhua) [reply source/shared/cpp/ObjectModel/ActionParserRegistration.h:10]: yes. but it's open to change. it's controlled by NamespaceIndentation. the default is none, and inner is used here. to add spaces, it has to be set to All. You can see the difference from https://clang.llvm.org/docs/ClangFormatStyleOptions.html

File: source/shared/cpp/ObjectModel/BaseCardElement.cpp (hunk 9 above)
Comment (licanhua) [reply source/shared/cpp/ObjectModel/BaseCardElement.cpp:30]: It's based on Paul's comments on /.clangformat, and achieved a balance between `BasedOnStyle: Microsoft` and the original `source/shared/cpp/.clang-format`

File: source/shared/cpp/ObjectModel/ChoiceSetInput.cpp (hunk 10 above)
Comment (licanhua) [reply source/shared/cpp/ObjectModel/ChoiceSetInput.cpp:12]: I have no idea. they both should have passed the formatting and verification.

File: source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer/ACRCustomSubmitTargetBuilder.h [hunk 11]
```diff
@@ -5,13 +5,9 @@
 //  Copyright © 2020 Microsoft. All rights reserved.
 //
 
-#import <Foundation/Foundation.h>
 #import <AdaptiveCards/AdaptiveCards.h>
+#import <Foundation/Foundation.h>
```
Comment (paulcam206) [inline source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer/ACRCustomSubmitTargetBuilder.h:9]: maybe @jwoo-msft knows more, but in UWP the order of `#include` (not `#import`) is important and shouldn't be sorted (order matters).

I know that the current `.clang-format` disables `#include` sorting, but does order matter in `#import` directives in ObjC?

File: source/experimental/ConsoleRenderer/ConsoleRenderer/ConsoleRender.cpp [hunk 12]
```diff
@@ -12,15 +12,16 @@
 using namespace AdaptiveCards;
 using namespace std::string_literals;
 
-size_t RenderElements(std::vector<std::shared_ptr<BaseCardElement>> elements, size_t columnWidth, size_t x, size_t y, std::vector<std::string>& consoleString);
-size_t RenderColumnSet(ColumnSet &set, size_t columnWidth, size_t x, size_t y, std::vector<std::string> &consoleString);
-size_t RenderColumn(Column &set, size_t columnWidth, size_t x, size_t y, std::vector<std::string> &consoleString);
-size_t RenderFactSet(FactSet &set, size_t FactWidth, size_t x, size_t y, std::vector<std::string> &consoleString);
-size_t RenderFact(Fact &set, size_t FactWidth, size_t x, size_t y, std::vector<std::string> &consoleString);
+size_t RenderElements(
```
Comment (paulcam206) [inline source/experimental/ConsoleRenderer/ConsoleRenderer/ConsoleRender.cpp:15]: We should just delete the `source/experimental` directory altogether :)

**Review by paulcam206 — APPROVED** [review]

File: source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer/ACRCustomSubmitTargetBuilder.h (hunk 11 above)
Comment (jwoo-msft) [reply source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer/ACRCustomSubmitTargetBuilder.h:9]: no, I never encountered issues after changing the order.

File: source/experimental/ConsoleRenderer/ConsoleRenderer/ConsoleRender.cpp (hunk 12 above)
Comment (licanhua) [reply source/experimental/ConsoleRenderer/ConsoleRenderer/ConsoleRender.cpp:15]: I can do it in a separate PR

**Review by almedina-ms — APPROVED** [review]
